corehq/motech/requests.py

```diff
@@ -292,3 +311,42 @@ def validate_user_input_url_for_repeaters(url, domain, src):
                 'reason': e.reason,
             })
             raise
+
+
+def apply_request_backoff(
```

Comment:
I decided to implement a kind of generic way to apply a backoff to functions making requests, hence implemented it as decorator for ease of use.